### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by oliverturner**
Webpack 2, Babel, ES6, re-architected data types, refactored Page class, more tests, small demo showing the fluent interface.

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  <script src="../dist/ticketmaster-client-1.0.0.min.js"></script>
+  <script>
+    (function(TMAPI) {
+      function logPage(page) {
+        console.log(page.page.number, page.items.map((item) => item.name))
+
+        return page.getNext()
+      }
+
+      const sdk = TMAPI.default('pTAf3X1DA6XCHjaQaw0WmlgcA0j1r32C')
```
Comment (adammeghji) [inline docs/index.html]: Is there a way we can remove this API key from the source, yet have it be clear to the user that this `docs/index.html` file requires you to make that change locally?

Comment (oliverturner) [inline docs/index.html]: Good idea. Done!

**Comment by adammeghji** [comment]
@oliverturner LGTM -- thanks for these amazing contributions!  Ready to merge in it's present state?

**Comment by oliverturner** [comment]
@adammeghji I think it is! One concern though: the Commerce API seems to require a channel id? Means I can't do any integration tests right now...

Other than that I think we're good to go :)

**Comment by adammeghji** [comment]
@oliverturner w.r.t. the Channel ID, might be a question best for @evohollywood.

Meanwhile, I'll merge this awesome body of work -- thanks again!

**Pull request merged**